### ISSUE TITLE
everybody needs support for buckets or regions :-)

### ISSUE THREAD

**Issue by DerDu** [memcached/memcached#157]
This is in fact a feature request ;-)

or it's me and it is already there?

--EDIT--
Just to be sure, i know about multiple instances, but if your not able to get them we need a native way to simulate this within one - lets say shared - instance

**Comment by dormando** [comment]
what exactly are you asking for?

**Comment by DerDu** [comment]
What i mean:

- i want to use **1** Memcached-Server running **1** Memcached-Instance, cause of web-hosting limitations
- the web application has different consumers "sharing" this Memcached-Server-Instance
- **BUT**: every consumer MUST HAVE it's own cache, because they're using identical keys to cache values so they would get values from other consumers == fail :-(

What i want:

- a kind of "option" to define seperated "regions" at connection setup
- so consumer A is routed to store key/values only in region A and consumer B only in region B in this single instance (e.g. bins in chouchdb)
- this would prevent consumer A getting values from consumer B

In short:

- to avoid naming collisions i need to partition key/value data with buckets like:
- http://docs.couchbase.com/developer/dev-guide-3.0/buckets.html
- http://docs.couchbase.com/developer/dev-guide-3.0/partition-data.html

**Comment by dormando** [comment]
Ok. The storage engine branch we had years ago could do this to a degree. Dustin had created a bucket engine which ran other engines underneath it... Unfortunately we weren't able to reconcile the branches.

It's quite a lot of work to get the current code to do that. It would take a series of patches and/or another attempt at a storage engine system. While it's definitely not everybody who needs it, I don't want to use that as an excuse. Instead I'll use the excuse of "I have limited time, no other active maintainers, and very few people asking for this sadly complicated feature" :) Hopefully someday we could get it.
